pkg/apis/build/v1alpha1/build_source.go

```diff
@@ -0,0 +1,17 @@
+// Copyright The Shipwright Contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package v1alpha1
+
+// BuildSourceType enumerates build source type names.
+type BuildSourceType string
+
+// BuildSource remote artifact definition.
+type BuildSource struct {
+	// Name instance entry.
+	Name string `json:"name"`
+
+	// URL remote artifact location.
```

Comment:
any restriction around not supporting auth credentials in the URL should be included in the godoc IMO

Reply:
bump @otaviof - I saw the auth restriction in the markdown down for this, but still IMO the godoc should note the restriction as well

update here to and then I'll add the approve label

Reply:
@gabemontero it's on the type declaration, o lines [10-11](https://github.com/otaviof/build/blob/build-40/pkg/apis/build/v1alpha1/build_source.go#L10-L11).

Reply:
ah ok sorry I missed that  ... was expecting it on the field

I will defer on the location

/approve

Between @ImJasonH who just commented, @HeavyWombat prior comments and making sure he is good, and I see @qu1queee wanted to review but was looking for the needs-rebase to be cleared up, we get those sorted out, squash commits, and then one of those folks can lgtm